File: output/mistralai_mistral_small_2603/zero/cpp/Torre.cpp

#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);

    int N;
    cin >> N;
    vector<vector<int>> board(N, vector<int>(N));
    vector<int> row_sums(N, 0);
    vector<int> col_sums(N, 0);

    for (int i = 0; i < N; ++i) {
        for (int j = 0; j < N; ++j) {
            cin >> board[i][j];
            row_sums[i] += board[i][j];
            col_sums[j] += board[i][j];
        }
    }

    int max_weight = 0;
    for (int i = 0; i < N; ++i) {
        for (int j = 0; j < N; ++j) {
            int weight = row_sums[i] + col_sums[j] - board[i][j];
            if (weight > max_weight) {
                max_weight = weight;
            }
        }
    }

    cout << max_weight << '\n';
    return 0;
}

File: output/deepseek_deepseek_v3_2/zero/python/Torre.py

import sys

def main():
    data = sys.stdin.read().strip().split()
    if not data:
        return
    idx = 0
    N = int(data[idx]); idx += 1
    board = []
    for _ in range(N):
        row = list(map(int, data[idx:idx+N]))
        idx += N
        board.append(row)
    
    row_sums = [0] * N
    col_sums = [0] * N
    
    for i in range(N):
        for j in range(N):
            row_sums[i] += board[i][j]
            col_sums[j] += board[i][j]
    
    max_weight = 0
    for i in range(N):
        for j in range(N):
            weight = row_sums[i] + col_sums[j] - 2 * board[i][j]
            if weight > max_weight:
                max_weight = weight
    
    print(max_weight)

if __name__ == "__main__":
    main()
